lib/blockchain.go

```diff
@@ -1782,11 +1799,11 @@ func (bc *Blockchain) processHeaderPoW(blockHeader *MsgDeSoHeader, headerHash *B
 	// index. If we're still syncing then it's safe to just set it. Otherwise, we
 	// need to make a copy first since there could be some concurrency issues.
 	if bc.isSyncing() {
-		bc.blockIndex[*newNode.Hash] = newNode
+		bc.addNewBlockNodeToBlockIndex(newNode)
 	} else {
 		newBlockIndex := bc.CopyBlockIndex()
-		newBlockIndex[*newNode.Hash] = newNode
-		bc.blockIndex = newBlockIndex
+		bc.blockIndexByHash = newBlockIndex
```

Comment:
If there are concurrency issues that affect the `blockIndexByHash`, then they would affect the blockIndexByHeight map too. Should you copy that second map as well?

FWIW, my read on the code is that the copy doesn't really protect against concurrency issues because the copy itself isn't thread safe (there's no additional lock around it). But the above comment applies for consistency

Reply:
Yeah I don't think this is actually doing  what its trying to do here.